common/include/pcl/correspondence.h

```diff
@@ -76,10 +77,20 @@ namespace pcl
     inline Correspondence () = default;
 
     /** \brief Constructor. */
-    inline Correspondence (int _index_query, int _index_match, float _distance) :
+    inline Correspondence (index_t _index_query, index_t _index_match, float _distance) :
       index_query (_index_query), index_match (_index_match), distance (_distance)
     {}
 
+    /** \brief Constructor. */
+    template <typename T = pcl::index_t,  std::enable_if_t<!std::is_same<T, int>::value, index_t> = 0>
+    #if defined(_MSC_VER)
+    inline
+    #endif
+    PCL_DEPRECATED(1, 13, "use  constructor that accepts index_t parameters instead")
+    Correspondence (int _index_query, int _index_match, float _distance) :
+      index_query (static_cast<index_t>(_index_query)), index_match (static_cast<index_t>(_index_match)), distance (_distance)
+    {}
+
```

Comment:
Do we need to deprecate this? `index_t` is int by default? Deprecation only comes into play if the arguments in the function signature representing indices are different than int. This is not one of those cases.

Reply:
Yeah, that makes sense. So I should remove the old constructor altogether?